libs/designsystem/form-field/src/form-field.component.scss

```
@@ -67,3 +68,46 @@ label {
     }
   }
 }
+
+// Background color needs to be synchronized with input.component
+// Background color must be fully opaque to cover native icon
+@supports selector(:has(+ *)) {
+  .affix-container:has(input[type='date']) {
+    --date-input-background-color: var(--kirby-inputs-background-color);
+
+    @include interaction-state.apply-hover {
+      --date-input-background-color: #e0e0e0;
+    }
+
+    @include interaction-state.apply-active {
+      --date-input-background-color: #cbcbcb;
+    }
+
+    // Take care of custom icon if it's present
```

I would suggest to explain what this does on one line and remove the rest of the comments in this block
```suggestion
    // Cover native calendar icon if custom icon is present
```

---

With the exception of 
`// Clicks etc. hit native icon/button instead of suffixed element`